apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -122,7 +122,19 @@ class DataFrame(TileDBArray, somacore.DataFrame):
         it must be ``None``.
     """
 
-    __slots__ = ()
+    __slots__ = "_first_write"
+
+    def __init__(
+        self,
+        handle: ArrayWrapper,
+        *,
+        _dont_call_this_use_create_or_open_instead: str = "unset",
+    ):
+        self._first_write = True
+        super().__init__(
+            handle,
+            _dont_call_this_use_create_or_open_instead="tiledbsoma-internal-code",
+        )
```

Comment:
@thetorpedodog I wanted your feedback on this addition I just added since you designed these classes.

On the first write to the DataFrame, we check if the PyArrow schema contains any dictionaries and update the TileDB ArraySchema with the enumeration values from the PyArrow array. I am keeping track of this by checking `self._first_write` which I now initialize in the `DataFrame` constructor. Upon first write, `self._first_write` gets toggled to False. I'm wondering if what I added here looks OK or if there's a better way to do what I'm trying to do.

Reply:
This seems like as good a way to go about this as I could think of. I might make a minor change to naming (maybe call it something like `_enumerations_written`, which goes from False to True?) but overall it makes sense and is reasonably simple.

I have one question which is more of a practical concern: how does this interact with it when there is existing data in the already-written part of the DataFrame? Do we just always assume that it existing data will have a compatible enum schema with new data? Since I am only passingly familiar with how enumerations work, I don’t have a strong opinion about this; basically provided this is something that you’ve thought about, whatever the answer is is fine by me.

Reply:
> how does this interact with it when there is existing data in the already-written part of the DataFrame?

That's an excellent point. I think I'll need something that checks whether this is the first write to a newly created DataFrame.